task/buildah-oci-ta/0.2/buildah-oci-ta.yaml

```diff
@@ -274,8 +274,12 @@ spec:
           echo "Cannot find Dockerfile $DOCKERFILE"
           exit 1
         fi
-        if [ -n "$JVM_BUILD_WORKSPACE_ARTIFACT_CACHE_PORT_80_TCP_ADDR" ] && grep -q '^\s*RUN \(./\)\?mvn' "$dockerfile_path"; then
-          sed -i -e "s|^\s*RUN \(\(./\)\?mvn\)\(.*\)|RUN echo \"<settings><mirrors><mirror><id>mirror.default</id><url>http://$JVM_BUILD_WORKSPACE_ARTIFACT_CACHE_PORT_80_TCP_ADDR/v1/cache/default/0/</url><mirrorOf>*</mirrorOf></mirror></mirrors></settings>\" > /tmp/settings.yaml; \1 -s /tmp/settings.yaml \3|g" "$dockerfile_path"
+
+        dockerfile_copy=/tmp/$(basename "$dockerfile_path")
```

Comment:
Tekton tasks run isolated in a Pod, so nothing except this script should have access to `/tmp` or any other part of the filesystem (except for the shared `workspaces`)

But it is something we'll need to address if we ever want to make the scripts CI-agnostic, so I will make the change

Reply:
In most cases this will be just /tmp/Dockerfile, which is more than guessable :D But is this really a concern? We are already putting several things there. Is it not the case that this area can be accessed only by the task itself?

Reply:
> In most cases this will be just /tmp/Dockerfile, which is more than guessable :D But is this really a concern? We are already putting several things there. Is it not the case that this area can be accessed only by the task itself?

I think this can be considered a good defensive programming technique which doesn't cost us anything and requires approximately 30s to fix, push and be done with it which is way shorter than it took me to read and respond to your comment :) .

Reply:
fixed